k8s/core/configmap/configmap_lock_v2.go

```diff
@@ -186,8 +186,10 @@ func (c *configMap) IsKeyLocked(key, requester string) (bool, string, error) {
 		c.kLocksV2Mutex.Lock()
 		lock := c.kLocksV2[key]
 		c.kLocksV2Mutex.Unlock()
-		lock.Lock()
-		defer lock.Unlock()
+		if lock != nil {
```

Comment:
if lock == nil {
it means it's not locked, let's return appropriate message
}

In the future if someone forgets to do a nil check before using lock, we'll be in another nil panic